src/com/qamp/HarisJasarevic/homeworks/homework5/task1/SavingsAccount.java

```diff
@@ -0,0 +1,54 @@
+package com.qamp.HarisJasarevic.homeworks.homework5.task1;
+
+public class SavingsAccount extends BankAccounts {
+
+    private double dailyTransactionLimit;
+    private int transactionsLeft;
+
+    protected SavingsAccount (final AccountOwner accountOwner, final String serialNumber, final int passCode, final int dailyTransactionLimit) {
+        super(accountOwner, serialNumber, passCode);
+        this.dailyTransactionLimit = 1000;
+        this.transactionsLeft = 3;
+    }
+
+    @Override
+
+    public double moneyWithdraw (final double withdrawAmount) throws IllegalStateException {
+        if (balance - withdrawAmount < 0) {
+            throw new IllegalStateException("There is no enough funds available!");
+        }
+        if (withdrawAmount > dailyTransactionLimit) {
+            throw new IllegalStateException("Withdrawal amount exceeds the daily transaction limit!");
+        }
+        if (transactionsLeft == 0) {
+            throw new IllegalStateException("You exceeded the limit of daily transactions!");
+        }
```

Comment:
Since this comes up in both withdraw and deposit you could consider creating an helper to perform the check